src/main/java/com/coveo/pushapiclient/ApiUrl.java

```diff
@@ -0,0 +1,106 @@
+package com.coveo.pushapiclient;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Private util class to extract dynamic parts from a Push API URL
+ *
+ * @See https://docs.coveo.com/en/1546#push-api-url
+ */
+class ApiUrl {
```

Comment:
Wondering if `PushApiUrl` would be a clearer name considering it's mentioned in the comment? Just `ApiUrl` feels a little vague,